src/scripts/main.js

```diff
@@ -1,3 +1,216 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('.game-header .button');
+
+button.addEventListener('click', handleStartClick);
+
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+
+const gameField = document.querySelector('.game-field');
+const rows = [...gameField.rows];
+const allCells = [...gameField.querySelectorAll('td')];
+
+function setGlobalHandleKeydown(keyCode) {
+  if (keyCode === 'ArrowUp') {
+    clickArrowUp();
+  }
+
+  if (keyCode === 'ArrowDown') {
+    clickArrowDown();
+  }
+
+  if (keyCode === 'ArrowRight') {
+    clickArrowRight();
+  }
+
+  if (keyCode === 'ArrowLeft') {
+    clickArrowLeft();
+  }
+}
+
+function addRandomCell() {
+  const emptyCells = allCells.filter(cell => cell.textContent.length === 0);
+
+  const randomCell = Math.floor(Math.random() * emptyCells.length);
+  const cellContent = Math.random() < 0.9 ? 2 : 4;
+
+  emptyCells[randomCell].innerText = cellContent;
```

Comment:
Consider using `textContent` instead of `innerText` for better performance and consistency. 🚀